pelican_image_process/__init__.py

```diff
@@ -10,7 +10,9 @@
 import collections
 import copy
 import functools
+import logging
 import os.path
+import posixpath
```

Comment:
posixpath is quite platform specific. I would like to see Pathlib's path related stuff instead. 

However consider to just keep os.path calls in the code instead of using posixpath. With this change we see a code base which is mixing different ways of dealing with paths and I would like avoid that we get a situation where the same thing is done using 3 different modules.

Reply:
In general, I'm a huge fan of `pathlib.Path` as well, but URL's are weird because they are sort of a cross between filepaths and strings. URL's are also weird because they follow posix path separator conventions, even on Windows.

So maybe `posixpath.join` make sense here?

Reply:
posixpath.py is platform-specific only in the sense that it handles POSIX paths. The module is available and acts the same way on [all CPython platforms](https://github.com/python/cpython/blob/master/Lib/posixpath.py). Paths in URLs are likely POSIX paths, but we'll have to agree to leave this claim a conjecture since I can't find any reliable source for it. :smiley:

I'd rather not port us on pathlib here, honestly. I'm quite sure I'd get it wrong. :sweat_smile:

Reply:
>  Paths in URLs are likely POSIX paths, but we'll have to agree to leave this claim a conjecture since I can't find any reliable source for it.

Go to the [source definition for URLS](https://www.w3.org/Addressing/URL/url-spec.txt). You'll find backslashes (`\`) listed as "national" characters that "do not appear in any productions and therefore may not appear in URLs".

Reply:
Right. But if backslashes did appear in the path parts, the prior behavior certainly didn't produce correct URLs either. :stuck_out_tongue: 

I guess the constituents might need to mapped through a [`urllib.parse.quote()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote) call.